### PR TITLE
Setup Kubevirt provider network controller in the seed-controller-manager

### PR DESCRIPTION
**Pull request by cnvergence**
**What this PR does / why we need it**:
This PR sets up a new controller that will reconcile clusters and configure KKP deployments with specific provider network requirements.
First use case will be setting up cluster isolation between customer networks.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Setup KubeVirt network controller in the seed-controller-manager.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: pkg/ee/kubevirt-network-controller/controller.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		workerName:   workerName,
+		recorder:     mgr.GetEventRecorderFor(ControllerName),
+	}
+
+	_, err := builder.ControllerManagedBy(mgr).
+		Named(ControllerName).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: numWorkers,
+		}).
+		For(&kubermaticv1.Cluster{}, builder.WithPredicates(predicateutil.Factory(func(o ctrlruntimeclient.Object) bool {
```
Comment (moadqassem) [inline pkg/ee/kubevirt-network-controller/controller.go:95]: Don't you need also to watch for subnets?

Comment (cnvergence) [reply pkg/ee/kubevirt-network-controller/controller.go:95]: As discussed earlier, we will drop this idea for now, as we will keep the subnets in the seed provider network settings

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Pull request merged**